plansys2_problem_expert/src/plansys2_problem_expert/ProblemExpert.cpp

```diff
@@ -248,6 +250,7 @@ ProblemExpert::existInstance(const std::string & name)
   while (!found && i < instances_.size()) {
     if (instances_[i].name == name) {
       found = true;
+      break; // breaking here speeds up this method
```

Comment:
I think that `found` var in the loop condition already speeds up the loop

Reply:
Same behavior should be added to `existAssignment `(for behavior homogeneity)

Reply:
I believe that all of the above has been addressed.